go/test/endtoend/cluster/cluster_process.go

```diff
@@ -282,6 +287,26 @@ func (cluster *LocalProcessCluster) ReStartVtgate() (err error) {
 	return err
 }
 
+// WaitForTabletsToHealthyInVtgate waits for all tablets in all shards to be healthy as per vtgate
+func (cluster *LocalProcessCluster) WaitForTabletsToHealthyInVtgate() {
+	var isRdOnlyPresent bool
+	for _, keyspace := range cluster.Keyspaces {
+		for _, shard := range keyspace.Shards {
+			isRdOnlyPresent = false
+			_ = cluster.VtgateProcess.WaitForStatusOfTabletInShard(fmt.Sprintf("%s.%s.master", keyspace.Name, shard.Name))
```

Comment:
Same here. It's better to return an error here and propagate it up the call stack.
And same comments below.